clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystemCore.java

```diff
@@ -34,91 +37,85 @@
-   * @throws InvalidPathException if the path is not a valid Tachyon path
-   * @throws IOException if the master is unable to create the file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
    */
-  long create(TachyonURI path, long blockSize, boolean recursive, long ttl) throws
-      BlockInfoException, FileAlreadyExistException, InvalidPathException, IOException;
+  long create(TachyonURI path) throws IOException, TachyonException;
+
+  long create(TachyonURI path, CreateOptions options) throws IOException, TachyonException;
 
   /**
-   * Deletes a file. If the file is a folder, its contents will be deleted recursively if the
-   * flag is set.
+   * Deletes a file or a directory.
    *
    * @param file the handler of the file to delete.
-   * @param recursive whether or not to delete all contents in a non empty folder
-   * @throws FileDoesNotExistException if the given file does not exist
-   * @throws IOException if the master is unable to delete the file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
    */
-  void delete(TachyonFile file, boolean recursive) throws FileDoesNotExistException, IOException;
+  void delete(TachyonFile file) throws IOException, TachyonException;
+
+  void delete(TachyonFile file, DeleteOptions options) throws IOException, TachyonException;
 
   /**
-   * Removes the file from Tachyon Storage. The underlying under storage system file will not be
-   * removed. If the file is a folder, its contents will be freed recursively if the flag is set.
+   * Removes the file from Tachyon, but not from UFS in case it exists there.
    *
    * @param file the handler for the file
-   * @param recursive whether or not to free all contents in a non empty folder
-   * @throws FileDoesNotExistException if the file does not exist
-   * @throws IOException if the master is unable to free the file for some other reason
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
    */
-  void free(TachyonFile file, boolean recursive) throws FileDoesNotExistException, IOException;
+  void free(TachyonFile file) throws IOException, TachyonException;
+
+  void free(TachyonFile file, FreeOptions options) throws IOException, TachyonException;
 
   /**
    * Gets the {@link FileInfo} object that represents the metadata of a Tachyon file.
    *
    * @param file the handler for the file.
    * @return the FileInfo of the file, null if the file does not exist.
-   * @throws IOException if the master cannot retrieve the file's metadata for some other reason
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
    */
-  FileInfo getInfo(TachyonFile file) throws IOException;
+  FileInfo getInfo(TachyonFile file) throws IOException, TachyonException;
```

Comment:
A general question, but should all of the methods in this file have options, in case there are new parameters in the future?